experimental/ModelBuilder/test/BenchMatMulVectorGPU.cpp

```diff
@@ -135,22 +140,124 @@ static SmallVector<linalg::ProcInfo, 2> getSubgroupIds(
   return procInfo;
 }
 
-void matMul(int m, int n, int k, int tileM, int tileN, int tileK,
-            bool correctness) {
+struct MatMulF32 {
+  using Type = float;
+  static mlir::Type getMLIRType(MLIRContext &ctx) {
+    return FloatType::getF32(&ctx);
+  }
+};
+
+struct MatMulI8 {
+  using Type = uint8_t;
+  static mlir::Type getMLIRType(MLIRContext &ctx) {
+    return IntegerType::get(8, &ctx);
+  }
+};
+
+struct MatMulI32 {
+  using Type = uint32_t;
+  static mlir::Type getMLIRType(MLIRContext &ctx) {
+    return IntegerType::get(32, &ctx);
+  }
+};
+
+// Class to emulate half float on CPU.
+class fp16 {
```

Comment:
THis is fine for now (and awesome that you implemented this yourself!). But I have seen this used in the past http://half.sourceforge.net/ .

Reply:
Ha I wish I knew about this, would have saved me some headache. Thanks!